core/src/test/java/org/elasticsearch/cluster/action/shard/ShardStateActionTests.java

```diff
@@ -143,6 +146,50 @@ public void onShardFailedFailure(DiscoveryNode master, TransportException e) {
         assertTrue(failure.get());
     }
 
+    public void testTimeout() throws InterruptedException {
+        final String index = "test";
+
+        clusterService.setState(stateWithStartedPrimary(index, true, randomInt(5)));
+
+        String indexUUID = clusterService.state().metaData().index(index).getIndexUUID();
+
+        AtomicBoolean progress = new AtomicBoolean();
+        assert !progress.get();
+        AtomicBoolean timedOut = new AtomicBoolean();
+        assert !timedOut.get();
+
+        TimeValue timeout = new TimeValue(3, TimeUnit.SECONDS);
```

Comment:
We don't need three seconds. Any positive time value *should* be okay. Reduced to one millisecond in 72c2bba6fe06127ed9dcaac84f074c8613bc6afe.